pkg/scheduler/algorithm/predicates/metadata.go

```diff
@@ -71,6 +77,15 @@ func RegisterPredicateMetadataProducer(predicateName string, precomp PredicateMe
 	predicateMetadataProducers[predicateName] = precomp
 }
 
+// RegisterPredicateMetadataProducerWithIgnoredExtendedResources registers a
+// PredicateMetadataProducer that creates predicate metadata with the provided
+// ignoredExtendedResources.
+func RegisterPredicateMetadataProducerWithIgnoredExtendedResources(ignoredExtendedResources sets.String) {
+	RegisterPredicateMetadataProducer("PredicateWithIgnoredExtendedResources", func(pm *predicateMetadata) {
```

Comment:
I changed the predicate name from `PodFitsResources` to `PredicateWithIgnoredExtendedResources`.

IIUC, the "precomp" function that sets the `ignoredExtendedResources` will override the one for the same predicate created here: https://github.com/kubernetes/kubernetes/blob/01bd3c4b743137f62b00d21ecd6f1559470de92f/pkg/scheduler/factory/plugins.go#L217

Is this the case? And I don't see why `predicateMetadataProducers` is not a slice here: https://github.com/kubernetes/kubernetes/blob/01bd3c4b743137f62b00d21ecd6f1559470de92f/pkg/scheduler/algorithm/predicates/metadata.go#L65